src/service/nestjs/src/api/channel/channel.gateway.ts

```diff
@@ -86,46 +83,44 @@ class ChannelGateway {
 		try {
 			const userId = socket.user.id;
 
-			await this.channelService.joinCheck(socket, joinData.channelId, userId);
-
 			const channel = await this.channelService.getChannel(joinData.channelId);
 			if (!channel) {
 				throw new BadRequestException('Channel not found');
 			}
-			if (await this.banService.isBanned(joinData.channelId, userId)) {
+			if (await this.banService.isBanned(channel.id, userId)) {
 				throw new BadRequestException('User is banned');
 			}
 			if (
 				channel.visibility === 'PROTECTED' &&
-				!(await this.channelService.validatePassword(joinData.channelId, joinData.password))
+				!(await this.channelService.validatePassword(channel.id, joinData.password))
 			) {
 				throw new BadRequestException('Wrong password');
 			}
 
-			const participant = await this.participantService.get(userId);
+			let participant = await this.participantService.getByUserId(userId);
 			if (participant) {
-				const newSocketId = plainToClass(ParticipantDto.Request.Update, { socketId: socket.id });
-				const error = await validate(newSocketId);
-
-				if (error.length > 0) {
-					throw new Error('Failed validation: ' + JSON.stringify(error));
+				if (participant.channelId !== channel.id) {
+					socket.leave(participant.channelId);
+					participant = await this.participantService.update(participant.id, {
+						channelId: channel.id,
```

Comment:
새로고침을 하고 들어오는 경우가 있을 수 있어서 `socketId`도 업데이트 될 수 있게 추가 부탁드려요
```suggestion
						channelId: channel.id,
						socketId: socket.id
```